src/mono/mono/mini/decompose.c

```diff
@@ -36,7 +36,10 @@ decompose_long_opcode (MonoCompile *cfg, MonoInst *ins, MonoInst **repl_ins)
 
 	switch (ins->opcode) {
 	case OP_LCONV_TO_I4:
-		ins->opcode = OP_SEXT_I4;
+		if (TARGET_SIZEOF_VOID_P == 4)
+			;
+		else
+			ins->opcode = OP_SEXT_I4;
```

Comment:
Wouldn't we only want this for non-LLVM 64-bit? `process_bb` already has a case for `OP_LCONV_TO_I4` that does what you'd expect.

Reply:
Not sure, didn't want to make larger changes close to the release.